### DISCUSSION TITLE
BAD Credential error

### DISCUSSION THREAD

**Discussion by urkadurka** [hassio-addons/addon-spotify-connect#269] [Q&A]
Hello,
At the plugin start the connection fails with this error 

[2024-09-05T08:55:51Z ERROR librespot] Connection failed: Login failed with reason: Bad credentials

Below the entire ERROR level log

----------------------------------------------------------
 Add-on: Spotify Connect
 Play Spotify music on your Home Assistant device
-----------------------------------------------------------
 Add-on version: 0.13.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 13.1  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2024.9.0
 Home Assistant Supervisor: 2024.08.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
Log level is set to ERROR
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun spotifyd (no readiness notification)
s6-rc: info: service legacy-services successfully started
[2024-09-05T08:55:51Z INFO  librespot] librespot 0.4.2 UNKNOWN (Built on 2024-01-06, Build ID: UYt0DULz, Profile: release)
[2024-09-05T08:55:51Z INFO  librespot_core::session] Connecting to AP "ap2-gew4.spotify.com:4070"
[2024-09-05T08:55:51Z ERROR librespot] Connection failed: Login failed with reason: Bad credentials
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service base-addon-log-level: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service base-addon-log-level successfully stopped
s6-rc: info: service base-addon-banner: stopping
s6-rc: info: service base-addon-banner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

This issue appeared recently, could you, please, suggest me ho to solve or which other information are required ? 

Thanks

**Comment by yeold** [comment]
Can confirm I have the same error.
A few minutes after that I get an email from spotify that they have reset my password due to "suspicious activity".
Seems like spotify changed something about their authentication process.

**Comment by nick-holmquist** [comment]
This appears to be because of a dependency in librespot:  https://github.com/librespot-org/librespot/issues/1308
There was a new release of librespot 3 days ago that seems to move to oauth vs username/password.  

This add-on will likely need to be updated to include the new librespot version and switch to oauth auth flow.

**Comment by aiakovou** [comment]
I have the same issue. The addon was started but immediately crashed.

-----------------------------------------------------------
 Add-on: Spotify Connect
 Play Spotify music on your Home Assistant device
-----------------------------------------------------------
 Add-on version: 0.13.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 14.0  (amd64 / qemux86-64)
 Home Assistant Core: 2024.12.2
 Home Assistant Supervisor: 2024.11.4
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun spotifyd (no readiness notification)
[09:52:12] INFO: Starting the Spotify daemon...
s6-rc: info: service legacy-services successfully started
[2024-12-11T07:52:12Z INFO  librespot] librespot 0.4.2 UNKNOWN (Built on 2024-01-06, Build ID: 8ZfE3H9w, Profile: release)
[2024-12-11T07:52:12Z WARN  librespot_core::apresolve] Ignoring blacklisted access point ap-gue1.spotify.com:4070
[2024-12-11T07:52:12Z WARN  librespot_core::apresolve] Ignoring blacklisted access point ap-gew4.spotify.com:443
[2024-12-11T07:52:12Z INFO  librespot_core::session] Connecting to AP "ap-gew1.spotify.com:4070"
[2024-12-11T07:52:13Z ERROR librespot] Connection failed: Login failed with reason: Bad credentials
[09:52:13] WARNING: spotifyd crashed, halting add-on
s6-rc: info: service legacy-services: stopping
[09:52:13] INFO: spotifyd stoped, restarting...
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service base-addon-log-level: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service base-addon-log-level successfully stopped
s6-rc: info: service base-addon-banner: stopping
s6-rc: info: service base-addon-banner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

**Comment by ericdpeterson** [comment]
Running into the same problem here.

-----------------------------------------------------------
 Add-on: Spotify Connect
 Play Spotify music on your Home Assistant device
-----------------------------------------------------------
 Add-on version: 0.15.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 16.0  (amd64 / generic-x86-64)
 Home Assistant Core: 2025.7.4
 Home Assistant Supervisor: 2025.07.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun spotifyd (no readiness notification)
[08:13:45] INFO: Starting the Spotify daemon...
s6-rc: info: service legacy-services successfully started
[2025-08-01T12:13:45Z INFO  librespot] librespot 0.6.0 VERGEN_IDEMPOTENT_OUTPUT (Built on 2025-06-27, Build ID: kVocG9Qo, Profile: release)
**[2025-08-01T12:13:45Z ERROR librespot] Invalid `--password` / `-p`: Password authentication no longer supported, use OAuth**
[08:13:45] WARNING: spotifyd crashed, halting add-on

**Reply by Helmi0407** [reply]
Same for me

**Comment by mckirkus** [comment]
You can remove the username and password from the configuration and it works.  If you're connecting to it from another device that has Spotify logged in it will use that for authentication.
